nodejs/eks/nodegroup.ts

```diff
@@ -310,17 +315,50 @@ ${customUserData}
 `;
         });
 
-    let amiId: pulumi.Input<string> = args.amiId!;
+    let amiId: any = (args.amiId) ? args.amiId : undefined;
+    const version: pulumi.Input<string> = args.version!;
     if (args.amiId === undefined) {
-        const eksWorkerAmi = aws.getAmi({
-            filters: [{
-                name: "name",
-                values: [ "amazon-eks-node-*" ],
-            }],
-            mostRecent: true,
-            owners: [ "602401143452" ], // Amazon
-        }, { parent: parent });
-        amiId = eksWorkerAmi.then(r => r.imageId);
+        if (version !== undefined) {
+            const eksWorkerAmi = aws.getAmiIds({
+                filters: [
+                    {
+                        name: "description",
+                        values: [ "*linux*", "*Linux*" ],
+                    },
+                    {
+                        name: "description",
+                        values: [ "*k8s*/bin/linux/amd64*"],
+                    },
+                    {
+                        name: "name",
+                        values: [ "amazon-eks-node-" + version + "*" ],
+                    },
+                ],
+                owners: [ "602401143452" ], // Amazon
+                sortAscending: true,
+            }, { parent: parent });
+            amiId = eksWorkerAmi.then(r => r.ids.pop());
+        } else {
+            const eksWorkerAmi = aws.getAmiIds({
+                filters: [
+                    {
+                        name: "description",
+                        values: [ "*linux*", "*Linux*" ],
+                    },
+                    {
+                        name: "description",
+                        values: [ "*k8s*/bin/linux/amd64*"],
+                    },
+                    {
+                        name: "name",
+                        values: [ "amazon-eks-node-*" ],
+                    },
+                ],
+                owners: [ "602401143452" ], // Amazon
+                sortAscending: true,
+            }, { parent: parent });
+            amiId = eksWorkerAmi.then(r => r.ids.pop());
```

Comment:
Given that this code is nearly identical but has finicky details in it - could we share better by just branching on whether to add `+ version` (or even just set version to `""` if not specified)?

Reply:
Yup will do! I was primarily focused on getting a fix out ASAP given the P0 status, but I'm certainly in favor of condensing this.

Reply:
Yeah - no need to block on this if it holds things up.